### ISSUE TITLE
Use WarpWallet with BIP39 code generation support

### ISSUE THREAD

**Issue by lirazsiri** [bitkey/bitkey#22]
There's a merge request for Warpwallet that adds BIP39 support. https://github.com/keybase/warpwallet/pull/22

If by the next release WarpWallet hasn't merged this, consider using this branch anyway. It would integrate nicely with Ian Coleman's Mnemonic Code generator tool and solve the privacy issues with address re-use.

**Comment by dabura667** [comment]
s/BIP39/mnemonic/

Without a guarantee of minimum 128 bits of entropy the feature does not fulfill BIP39.

"BIP39 import compatible mnemonics" would also be fine.

**Comment by lirazsiri** [comment]
Fair point, I changed the title. I think it would have been better to separate the entropy gathering specification to another BIP.  You can verify conformity to mnemonics and checksums, but it's really hard to verify how much entropy you're getting out of a blackbox RNG. You can hope your faithful computer is running a faithfully compiled binary of a CSPRNG who's source code you audited to be cryptographically secure, and you can hope the CSPRNG actually managed to collect 128-bits of entropy before you started using it, but as far as I can tell short of throwing unloaded dice yourself there's no way to guarantee you're getting that much entropy. Even then if you wanted an at least 128 bits of entropy you would need to add additional rolls of the dice and/or verify it's fairness.

This is a particularly sensitive issue for BitKey. Does anyone really know how to measure let alone guarantee that your computer randomly ends up in one of at least 2 ^ 128 states within N minutes of booting the OS, before you start seeding your CSPRNG?

**Comment by lirazsiri** [comment]
@dabura667 Your comment illustrates a problem with BIP39, or at least your interpretation of it. There is no well defined way to require 128 bits of entropy.

I added a comment to the BIP39 proposal here:

https://github.com/bitcoin/bips/wiki/Comments:BIP-0039

I and also opened a related BitKey feature request regarding recommending deterministic wallet creation:

https://github.com/bitkey/bitkey/issues/26
